### PR TITLE
Add postCreateCommand to the feature

### LINKED ISSUE #9: Install the postcreate script in the julia Feature
I believe that the `postcreate.jl` script referenced from the current Dev Container Template can be installed with the julia Feature.

https://github.com/microsoft/vscode-dev-containers/blob/8918c936dfaf3b03b923ce63c524adf52d32e492/containers/julia/.devcontainer/devcontainer.json#L17
https://github.com/julia-vscode/julia-devcontainer/blob/60c2852703d5f4e2fc419198a95c49be37d12ccb/postcreate.jl

That would be helpful when creating a new Dev Container Template using the julia Feature.

It may be a good idea to change the installation path, see also how the common-utils Feature copies scripts into the container.

https://github.com/devcontainers/features/blob/a9a7a25d60585ce2145374ec6825cbdc7f16cb90/src/common-utils/main.sh#L487

### PR DESCRIPTION
**Pull request by gerlero**
Closes #9 

Adds a [`postcreate.jl`](https://github.com/julia-vscode/julia-devcontainer/blob/4d501096eff958846dc04c44c5fb82d690c5605a/postcreate.jl) script (taken from https://github.com/julia-vscode/julia-devcontainer) to the container on install and sets is as the `postCreateCommand` of the feature.

I'm using `/usr/local/julia-devcontainer-features/postcreate.jl` for the location of the script inside the container, but I don't know the conventions here, and there might very well be a better path.

### CODE REVIEW

File: src/julia/devcontainer-feature.json
Comment (eitsupi) [inline src/julia/devcontainer-feature.json:21]: How about use object like `"julia-activate": "/usr/local/julia-devcontainer-features/postcreate.jl"`?
https://containers.dev/implementors/spec/#parallel-exec

File: src/julia/install.sh
Comment (eitsupi) [inline src/julia/install.sh:65]: I think it makes more sense to make the script included in the Feature executable than to do this here.

**Review by eitsupi — COMMENTED** [review]
Thanks for working on this!

A few minor comments.

**Review by eitsupi — CHANGES_REQUESTED** [review]
Please bump the minor version here.
https://github.com/JuliaLang/devcontainer-features/blob/45c7a4d2c7ae4f451ddc1692705e15ce931c43dc/src/julia/devcontainer-feature.json#L4

File: src/julia/install.sh
Comment (gerlero) [reply src/julia/install.sh:65]: I took that from https://github.com/devcontainers/features/blob/a9a7a25d60585ce2145374ec6825cbdc7f16cb90/src/common-utils/main.sh#L488. At least when trying out the feature from my local filesystem, looks like the `chmod +x` is not necessary even if the source file was not executable (I agree though that it should be executable).

I'll make `postcreate.jl` executable in the repo, and drop the `chmod` at line 65.

**Review by eitsupi — APPROVED** [review]
Looks good, thanks!

File: src/julia/devcontainer-feature.json
Comment (eitsupi) [inline src/julia/devcontainer-feature.json:23]: Thinking again, would this be appropriate with `postCreateCommand`, which is executed only once when the container is created.
`postStartCommand` or `postAttachCommand` would be executed each time the container is started
https://containers.dev/implementors/json_reference/

Comment (gerlero) [reply src/julia/devcontainer-feature.json:23]: I think running the script when the container is created should be enough? Testing with a package, once ``Pkg.develop()`` is called on container creation, that seems to persist across sessions.

Comment (eitsupi) [reply src/julia/devcontainer-feature.json:23]: Ok, thanks for checking this!

**Comment by eitsupi** [comment]
@JuliaLang @davidanthoff Could you take a look at this? (Unfortunately I cannot merge this.)

**Comment by gerlero** [comment]
> @JuliaLang @davidanthoff Could you take a look at this? (Unfortunately I cannot merge this.)

Soft ping to the maintainers—can we get this in?

**Review by davidanthoff — APPROVED** [review]
Looks good to me, sorry for being slow.

**Comment by davidanthoff** [comment]
Do we need to do something so that this is picked up? Tag or something? Or is it enough to have this on `main`?

**Comment by eitsupi** [comment]
> Do we need to do something so that this is picked up? Tag or something? Or is it enough to have this on `main`?

I assume this was released because you ran the CI manually. Thanks!

**Comment by davidanthoff** [comment]
Yep, I think so!

**Comment by gerlero** [comment]
Thanks! Yes, can confirm it's been released.

**Pull request merged**
